Submitted By: Bruce Dubbs <bdubbs@linuxfromscratch.org>
Date: 2011-10-10
Initial Package Version: 2.6.1
Origin: Upstream
Description: Patch two testsuite files due to a change in locale information for 
the ru local in binutils-2.14.

diff -Naur gcc-4.6.1.orig/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-1.cc gcc-4.6.1/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-1.cc
--- gcc-4.6.1.orig/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-1.cc	2010-09-15 05:46:39.000000000 -0500
+++ gcc-4.6.1/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-1.cc	2011-10-10 15:19:03.000000000 -0500
@@ -1,6 +1,6 @@
 // { dg-require-namedlocale "ru_RU.ISO-8859-5" }
 
-// Copyright (C) 2010 Free Software Foundation
+// Copyright (C) 2010, 2011 Free Software Foundation
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
@@ -49,7 +49,11 @@
   // get_weekday(iter_type, iter_type, ios_base&, 
   //             ios_base::iostate&, tm*) const
 
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 14)
+  iss.str("\xbf\xdd\x2e");
+#else
   iss.str("\xbf\xdd\xd4");
+#endif
   iterator_type is_it01(iss);
   tm time01;
   memset(&time01, -1, sizeof(tm));
@@ -67,7 +71,11 @@
   VERIFY( time02.tm_wday == 1 );
   VERIFY( errorstate == ios_base::eofbit );
 
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 14)
+  iss.str("\xbf\xdd\x2e\xd5\xd4\xd5\xdb\xec\xdd\xd8\xda");
+#else
   iss.str("\xbf\xdd\xd4\xd5\xd4\xd5\xdb\xec\xdd\xd8\xda");
+#endif
   iterator_type is_it03(iss);
   tm time03;
   memset(&time03, -1, sizeof(tm));
diff -Naur gcc-4.6.1.orig/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-2.cc gcc-4.6.1/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-2.cc
--- gcc-4.6.1.orig/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-2.cc	2010-09-15 05:46:39.000000000 -0500
+++ gcc-4.6.1/libstdc++-v3/testsuite/22_locale/time_get/get_weekday/char/38081-2.cc	2011-10-10 15:20:58.000000000 -0500
@@ -2,7 +2,7 @@
 
 // 2010-01-05  Paolo Carlini  <paolo.carlini@oracle.com>
 
-// Copyright (C) 2010 Free Software Foundation
+// Copyright (C) 2010, 2011 Free Software Foundation
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
@@ -50,6 +50,15 @@
   // get_weekday(iter_type, iter_type, ios_base&, 
   //             ios_base::iostate&, tm*) const
 
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 14)
+  const char* awdays[7] = { "\u0412\u0441\u002E",
+            "\u041F\u043D\u002E",
+            "\u0412\u0442\u002E",
+            "\u0421\u0440\u002E",
+            "\u0427\u0442\u002E",
+            "\u041F\u0442\u002E",
+            "\u0421\u0431\u002E" };
+#else
   const char* awdays[7] = { "\u0412\u0441\u043A",
 			    "\u041F\u043D\u0434",
 			    "\u0412\u0442\u0440",
@@ -57,6 +66,7 @@
 			    "\u0427\u0442\u0432",
 			    "\u041F\u0442\u043D",
 			    "\u0421\u0431\u0442" };
+#endif
 
   for (int i = 0; i < 7; ++i)
     {
